### ISSUE TITLE
Running app as root breaks on Wayland

### ISSUE THREAD

**Issue by flipflop97** [mmstick/systemd-manager#48]
The application launcher starts this app as root, which should be avoided, as it breaks the app on Wayland. Systemd can handle root requests itself when needed. I suggest changing the app launcher/.desktop to just start it with `systemd-manager` instead of `pkexec systemd-manager` (aliased as `systemd-manager-pkexec`).

**Comment by Cherkah** [comment]
hi,
what should i do to run the app with gnome wayland?

**Comment by flipflop97** [comment]
@Cherkah If you have a recent version of Gnome it should use Wayland by default, otherwise you can select `Gnome` instead of `Gnome on Xorg` on the login screen. If you run `echo $XDG_SESSION_TYPE` in your terminal it should say if you're using Wayland or X.

If that says Wayland then this app will also run natively on Wayland.

**Comment by Cherkah** [comment]
ok,
 it's my case (debian stretch testing...) so gnome with wayland by default.
as the rustup process do not work with waykand i'll remove rustup and install systemd-manager via the installation script so as to get a proper integration with PolicyKit.
lets a try and confirme next

**Comment by Cherkah** [comment]
> If that says Wayland then this app will also run natively on Wayland.

sorry but the app do not run on wayland. after the password nothing append.

**Comment by flipflop97** [comment]
Yes, that's exactly the problem. Apps can't run as root on Wayland. Only actions that need root access should request root permissions.

**Comment by mmstick** [comment]
> Systemd can handle root requests itself when needed.

This is not true. You are referring to the systemctl command. This application is not using systemctl, but communicating directly to systemd via it's dbus IPC API. Attempting to tell systemd to enable/disable a service will give you an `interactive authentication required` error.

```

**Comment by ignatenkobrain** [comment]
This is main blocker from getting sd-manager in Fedora (because there is wayland by default).

**Comment by keszybz** [comment]
>  This application is not using systemctl, but communicating directly to systemd via it's dbus IPC API. 

This is exactly what systemctl does. systemctl is not privileged, so anything that systemctl does can be independently performed from any dbus client.
